plugins/GSdx/GSUtil.cpp

```diff
@@ -396,6 +396,39 @@ D3D_FEATURE_LEVEL GSUtil::CheckDirect3D11Level(IDXGIAdapter *adapter, D3D_DRIVER
 	return SUCCEEDED(hr) ? level : (D3D_FEATURE_LEVEL)0;
 }
 
+GSRendererType GSUtil::GetBestRenderer()
+{
+	CComPtr<IDXGIAdapter1> adapter;
+	DXGI_ADAPTER_DESC1 desc;
+	CComPtr<IDXGIFactory1> DXGI;
+
+	if (GSUtil::CheckDXGI())
+	{
+		CreateDXGIFactory1(__uuidof(IDXGIFactory1), (void**)&DXGI);
```

Comment:
This function can fail, so you do need to check the return value.

Note: `CreateDXGIFactory1(IID_PPV_ARGS(&DXGI));` is an alternative.

Note 2: Variable naming consistency is somewhat absent in this function.